app/services/ValidatingPageBuilder.scala

```diff
@@ -165,4 +182,55 @@ class ValidatingPageBuilder @Inject() (pageBuilder: PageBuilder){
         }
     }
 
+  private def checkForSequencePageReuse(connections: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
+    val vertexMap = connections.map(pv => (pv.id, pv)).toMap
+    val mainFlow: List[String] = mainFlowPageIds(vertexMap)
+    val sequencePageIds = for{
+      pv <- connections.filterNot(_.flows.isEmpty)              // Sequences
+      flw <- pv.flows                                           // Flow Ids
+      p <- pageGraph(findPages(List(flw)), vertexMap, mainFlow) // Pages below sequecnes
+    } yield p.id
+
+    sequencePageIds.groupBy(x => x).toList.collect{case m if m._2.length > 1 => PageOccursInMultiplSequenceFlows(m._1)}
+  }
+
+  private def mainFlowPageIds(vertexMap: Map[String, PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[String] =
+    pageGraph(List(Process.StartStanzaId), vertexMap).map(_.id)
+
+  @tailrec
+  // Given a list of stanza ids, find all connected pages (wont follow links)
+  private def findPages(ids: List[String], seen: List[String] = Nil, acc: List[String] = Nil)(implicit stanzaMap: Map[String, Stanza]): List[String] =
```

Comment:
I would name this findPageIds because it is returning a list pf page ids.

Reply:
done